pkg/planner/core/task.go

```diff
@@ -2132,7 +2127,7 @@ func RemoveUnnecessaryFirstRow(
 				}
 			}
 			if canOptimize {
-				partialSchema.Columns = append(partialSchema.Columns[:partialCursor], partialSchema.Columns[partialCursor+1:]...)
```

Comment:
@Rustin170506 could it be a hidden bug here, if `canOptimize` is true for more than one iterations? I.e. since it loops over a fixed number, and here it will decrease the number of columns by one.

Reply:
No, I don't think we have this problem here, because we directly pass the partialSchema pointer. So I expect most references to point to the schema itself, rather than to the column slice.
We cloned the schema before passing it to the RemoveUnnecessaryFirstRow function.